### PR TITLE
507 product category list block

### LINKED ISSUE #507: [Product Home] Product Category List block
Create a Product Category List block and UI

https://www.figma.com/file/fgBV6T1rkYBvzQYzwwUnFt/Bioscience-hub-3?type=design&node-id=19011-234971&mode=design&t=2kBAcLaUMaXZ90KY-0

image

### PR DESCRIPTION
**Pull request by rgravitvl**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #507 

Test URLs:
- Before: https://main--danaher-ls-aem--hlxsites.hlx.page/us/en/product1
- After: https://507-product-category-list-block--danaher-ls-aem--hlxsites.hlx.page/us/en/product1
- After: https://507-product-category-list-block--danaher-ls-aem--hlxsites.hlx.page/us/en/product2

### CODE REVIEW

File: blocks/product-category-list/product-category-list.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  let products = await ffetch('/us/en/products-index.json')
+    .filter(({ title }) => title !== '')
+    .filter(({ image }) => image !== '')
+    .all();
+
+  const activeTagFilter = getSelectionFromUrl('tag');
+  let filteredProducts = products;
+  products = products.sort((item1, item2) => item1.title.localeCompare(item2.title));
+  if (activeTagFilter) {
```
Comment (mhaack) [inline blocks/product-category-list/product-category-list.js:21]: Additionally you need to filter for type = Category to show only categories. Right know also topic pages are shown.

Plus you only need top level categories.

**Review by mhaack — CHANGES_REQUESTED** [review]

File: blocks/product-category-list/filter.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import {
+  a, div,
+} from '../../scripts/dom-builder.js';
+import { toClassName } from '../../scripts/lib-franklin.js';
+
+const createFilters = (items, activeTag) => {
```
Comment (mhaack) [inline blocks/product-category-list/filter.js:6]: Is there a plan to us this for the blog/news hub as well?

**Review by mhaack — APPROVED** [review]

File: blocks/product-category-list/filter.js (hunk 2 above)
Comment (rgravitvl) [reply blocks/product-category-list/filter.js:6]: We can use

File: blocks/product-category-list/miniCard.js
[the remark this answers is not in the document]
Comment (rgravitvl) [reply blocks/product-category-list/miniCard.js:1]: done

**Review by davenichols-DHLS — CHANGES_REQUESTED** [review]
Hi @rgravitvl , sorry I have noticed that if I tag multiple OpCos then the filter breaks:

image

**Review by davenichols-DHLS — APPROVED** [review]
Approved - we will raise a task about anchoring down to the component when you click on it.

**Pull request merged**